go-controller/pkg/metrics/master.go

```diff
@@ -503,50 +523,75 @@ func NewControlPlaneRecorder() *ControlPlaneRecorder {
 	}
 }
 
-//Run will register the necessary metrics and add event handlers.
-func (ps *ControlPlaneRecorder) Run(sbClient libovsdbclient.Client) {
+//Run will manage metrics for monitoring control plane objects and block until stop signal received.
+//Monitors pod setup latency
+func (cpr *ControlPlaneRecorder) Run(sbClient libovsdbclient.Client, stop <-chan struct{}) {
 	// only register the metrics when we want them
 	prometheus.MustRegister(metricFirstSeenLSPLatency)
 	prometheus.MustRegister(metricLSPPortBindingLatency)
 	prometheus.MustRegister(metricPortBindingUpLatency)
 	prometheus.MustRegister(metricPortBindingChassisLatency)
+	cpr.queue = workqueue.New()
+	cpr.queueReady = true
 
 	sbClient.Cache().AddEventHandler(&cache.EventHandlerFuncs{
 		AddFunc: func(table string, model model.Model) {
 			if table != portBindingTable {
 				return
 			}
-			go ps.AddPortBindingEvent(model)
+			if cpr.queueReady {
+				cpr.queue.Add(item{op: addPortBinding, old: &model})
+			}
 		},
 		UpdateFunc: func(table string, old model.Model, new model.Model) {
 			if table != portBindingTable {
 				return
 			}
-			go ps.UpdatePortBindingEvent(old, new)
+			if cpr.queueReady {
+				cpr.queue.Add(item{op: updatePortBinding, old: &old, new: &new})
+			}
 		},
 		DeleteFunc: func(table string, model model.Model) {
 		},
 	})
+
+	go func() {
+		wait.Until(cpr.runWorker, queueCheckPeriod, stop)
+		cpr.queueReady = false
+		cpr.queue.ShutDown()
+	}()
+}
+
+func (cpr *ControlPlaneRecorder) AddPodEvent(podUID kapimtypes.UID) {
+	if cpr.queueReady {
+		cpr.queue.Add(item{op: addPod, podUID: podUID})
+	}
 }
 
-func (ps *ControlPlaneRecorder) AddPodEvent(podUID kapimtypes.UID) {
-	ps.Lock()
-	ps.podRecords[podUID] = &record{timestamp: time.Now(), timestampType: firstSeen}
-	ps.Unlock()
+func (cpr *ControlPlaneRecorder) addPodEvent(podUID kapimtypes.UID) {
+	cpr.podRecords[podUID] = &record{timestamp: time.Now(), timestampType: firstSeen}
+}
+
+func (cpr *ControlPlaneRecorder) CleanPodRecord(podUID kapimtypes.UID) {
+	if cpr.queueReady {
+		cpr.queue.Add(item{op: cleanPod, podUID: podUID})
+	}
```

Comment:
I don't know why but function names looked a bit backwards to me.

s/addPodEvent/addPodRecord
s/CleanPodRecord/CleanPodEvent

WDYT?

Reply:
Theyre not named good..

I am tempted to get rid of Record and Event suffixes from all the methods. By calling oc.metricsRecorder.AddPod, it should be clear what its purpose is.